url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/syntax/builder/SyntaxBuilderParserTables.java

```diff
@@ -0,0 +1,720 @@
+		g.put("`stringdef`", new Integer(17));
+		g.put("`identifier`", new Integer(16));
+		g.put("sequence", new Integer(9));
+		g.put("`ruleref`", new Integer(11));
+	}
+	private void loadGoto_22()	{
+		Hashtable g = new Hashtable(1, 1);
+		gotoTable.add(g);
+		g.put("`bnf_chardef`", new Integer(30));
+	}
+	private void loadGoto_25()	{
+		Hashtable g = new Hashtable(2, 1);
+		gotoTable.add(g);
+		g.put("intersectionsubtract", new Integer(31));
+		g.put("'-'", new Integer(26));
+	}
+	private void loadGoto_26()	{
+		Hashtable g = new Hashtable(7, 1);
+		gotoTable.add(g);
+		g.put("`identifier`", new Integer(16));
+		g.put("`bnf_chardef`", new Integer(7));
+		g.put("`ruleref`", new Integer(11));
+		g.put("intersectionunit", new Integer(33));
+		g.put("set", new Integer(19));
+		g.put("intersectionstartunit", new Integer(32));
+		g.put("`stringdef`", new Integer(17));
+	}
+	private void loadGoto_28()	{
+		Hashtable g = new Hashtable(2, 1);
+		gotoTable.add(g);
+		g.put("')'", new Integer(34));
+		g.put("'|'", new Integer(21));
+	}
+	private void loadParseActionTable()	{
+		parseTable = new ArrayList(35);
+
+		loadParseAction_0();
+		loadParseAction_1();
+		loadParseAction_2();
+		loadParseAction_3();
+		loadParseAction_4();
+		loadParseAction_5();
+		loadParseAction_6();
+		loadParseAction_7();
+		loadParseAction_8();
+		loadParseAction_9();
+		loadParseAction_10();
+		loadParseAction_11();
+		loadParseAction_12();
+		loadParseAction_13();
+		loadParseAction_14();
+		loadParseAction_15();
+		loadParseAction_16();
+		loadParseAction_17();
+		loadParseAction_18();
+		loadParseAction_19();
+		loadParseAction_20();
+		loadParseAction_21();
+		loadParseAction_22();
+		loadParseAction_23();
+		loadParseAction_24();
+		loadParseAction_25();
+		loadParseAction_26();
+		loadParseAction_27();
+		loadParseAction_28();
+		loadParseAction_29();
+		loadParseAction_30();
+		loadParseAction_31();
+		loadParseAction_32();
+		loadParseAction_33();
+		loadParseAction_34();
+	}
+	private void loadParseAction_0()	{
+		Hashtable p = new Hashtable(1, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(-2));
+	}
+	private void loadParseAction_1()	{
+		Hashtable p = new Hashtable(1, 1);
+		parseTable.add(p);
+		p.put("\"::=\"", new Integer(-2));
+	}
+	private void loadParseAction_2()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(2));
+		p.put("\"EoI\"", new Integer(2));
+	}
+	private void loadParseAction_3()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(-2));
+		p.put("\"EoI\"", new Integer(0));
+	}
+	private void loadParseAction_4()	{
+		Hashtable p = new Hashtable(7, 1);
+		parseTable.add(p);
+		p.put("'|'", new Integer(22));
+		p.put("'('", new Integer(-2));
+		p.put("';'", new Integer(22));
+		p.put("`bnf_chardef`", new Integer(-2));
+		p.put("`ruleref`", new Integer(-2));
+		p.put("`identifier`", new Integer(-2));
+		p.put("`stringdef`", new Integer(-2));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 367:375, 490:498.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.